`bin/ci build` fails on travis CI 64-bit Linux

`bin/ci build` fails on travis CI 64-bit Linux, it seems like it is missing libgc.
```
CRYSTAL_CONFIG_PATH=`pwd`/src ./bin/crystal build  -o .build/crystal src/compiler/crystal.cr -D without_openssl -D without_zlib
./bin/crystal build  -o .build/std_spec spec/std_spec.cr
Using compiled compiler at `.build/crystal'
/usr/bin/ld: cannot find -lgc
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o '/mnt/.build/std_spec'  -rdynamic  -lyaml `/usr/bin/llvm-config-3.8 --libs --system-libs --ldflags 2> /dev/null` -lstdc++ `command -v pkg-config > /dev/null && pkg-config --libs libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs libcrypto || printf %s '-lcrypto'` -lreadline -lgmp -lz -lxml2 -lpcre -lm -lgc -lpthread /mnt/src/ext/libcrystal.a -levent -lrt -ldl -L/usr/lib -L/usr/local/lib`
Makefile:109: recipe for target '.build/std_spec' failed
make: *** [.build/std_spec] Error 1
on_os linux docker run --rm -t -u 2000 -v /home/travis/build/crystal-lang/crystal:/mnt -w /mnt -e LIBRARY_PATH=/opt/crystal/embedded/lib/ -e CRYSTAL_CACHE_DIR=/tmp/crystal jhass/crystal-build-x86_64 linux64 /bin/bash -c 'make crystal std_spec compiler_spec docs' exited with 2
```
[Full travis log](https://travis-ci.org/crystal-lang/crystal/jobs/321499601)

This is not a singular incident but happens across several PRs ([another example](https://travis-ci.org/crystal-lang/crystal/jobs/321596895)).

Builds on 32-bit Linux as well as Mac OS builds on Circle-CI are not affected.

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,7 +9,7 @@ services:
 matrix:
   fast_finish: true
   include:
-    - env: ARCH=i386 ARCH_CMD=linux32
+    - env: ARCH=i386 ARCH_CMD=linux32 ARCH_LIBRARY_PATH=/opt/crystal/embedded/lib/
       os: linux
     - env: ARCH=x86_64 ARCH_CMD=linux64 DEPLOY=true
       os: linux

--- a/bin/ci
+++ b/bin/ci
@@ -164,7 +164,7 @@ with_build_env() {
     -u $(id -u) \
     -v $(pwd):/mnt \
     -w /mnt \
-    -e LIBRARY_PATH="/opt/crystal/embedded/lib/" \
+    -e LIBRARY_PATH="${ARCH_LIBRARY_PATH:-/usr/lib/crystal/lib/}" \
     -e CRYSTAL_CACHE_DIR="/tmp/crystal" \
     "jhass/crystal-build-$ARCH" \
     "$ARCH_CMD" /bin/bash -c "'$command'"

